registry/dogu_etcd.go

```diff
@@ -0,0 +1,241 @@
+// Gets the dogu from v2 registry. If dogu is not installed in v2 registry, v1 registry is used.
+func (reg *combinedEtcdDoguRegistry) Get(name string) (*core.Dogu, error) {
+	v2dogu, err := reg.v2DoguRegistry.Get(name)
+	if err != nil {
+		if !IsKeyNotFoundError(err) {
+			return nil, errors.Wrap(err, "could not get dogu from v2 registry")
+		}
+
+		return reg.v1DoguRegistry.Get(name)
+	}
+
+	return v2dogu, nil
+}
+
+// GetAll returns all registered dogus
+// Collects all dogus in v1 as well as in v2 registry.
+func (reg *combinedEtcdDoguRegistry) GetAll() ([]*core.Dogu, error) {
+	var allDogus []*core.Dogu
+
+	v2dogus, err := reg.v2DoguRegistry.GetAll()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get all v2 dogus")
+	}
+
+	v1dogus, err := reg.v1DoguRegistry.GetAll()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get all v1 dogus")
+	}
+
+	allDogus = append(allDogus, v2dogus...)
+
+	for _, v1dogu := range v1dogus {
+		if !core.ContainsDoguWithName(v2dogus, v1dogu.Name) {
+			allDogus = append(allDogus, v1dogu)
+		}
+	}
+
+	return allDogus, nil
+}
+
+// Unregister removes a dogu from the registry
+// Removes in v1 as well as in v2 registry.
+func (reg *combinedEtcdDoguRegistry) Unregister(name string) error {
+	err := reg.v1DoguRegistry.Unregister(name)
+	if err != nil {
+		return errors.Wrap(err, "could not unregister v1 dogu")
+	}
+
+	err = reg.v2DoguRegistry.Unregister(name)
+	if err != nil && !IsKeyNotFoundError(err) {
+		return errors.Wrap(err, "could not unregister v2 dogu")
+	}
+
+	return nil
+}
+
+// IsEnabled returns true if the dogu is installed and enabled.
+// Use v1 registry as it contains v1 as well as converted v2 dogus.
+func (reg *combinedEtcdDoguRegistry) IsEnabled(name string) (bool, error) {
+	return reg.v1DoguRegistry.IsEnabled(name)
+}
+
+type etcdDoguRegistry struct {
+	path           string
+	client         *resilentEtcdClient
+	formatProvider core.DoguFormatProvider
+}
+
+// Enable enables a specific dogu in the backend. The method will create the
+// current key to the dogu path
+func (reg *etcdDoguRegistry) Enable(dogu *core.Dogu) error {
+	core.GetLogger().Infof("enable dogu %s:%s", dogu.GetSimpleName(), dogu.Version)
+
+	path := reg.path + "/" + dogu.GetSimpleName() + "/current"
+	core.GetLogger().Debug("set etcd value at", path)
+	_, err := reg.client.Set(path, dogu.Version, nil)
+	return err
+}
+
+// Register registers the dogu at the registry backend
+func (reg *etcdDoguRegistry) Register(dogu *core.Dogu) error {
+	// convert dogu to json
+	data, err := reg.formatProvider.WriteDoguToString(dogu)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal dogu")
+	}
+
+	core.GetLogger().Infof("register dogu %s:%s", dogu.GetSimpleName(), dogu.Version)
+
+	// register dogu as json on etcd
+	path := reg.path + "/" + dogu.GetSimpleName() + "/" + dogu.Version
+	core.GetLogger().Debug("set etcd value on", path)
+	_, err = reg.client.Set(path, string(data), nil)
```

Comment:
Redundant type conversion from string to string
```suggestion
	_, err = reg.client.Set(path, data, nil)
```

- [ ] done